compiler/rustc_query_impl/src/plumbing.rs

```diff
@@ -655,6 +700,21 @@ macro_rules! define_queries {
             }
         }
 
+        const TRY_COLLECT_ACTIVE_JOBS: &[for<'tcx> fn(TyCtxt<'tcx>, &mut QueryMap<DepKind>)] =
+            &[$(query_impl::$name::try_collect_active_jobs),*];
+
+        const ALLOC_SELF_PROFILE_QUERY_STRINGS: &[
+            for<'tcx> fn(TyCtxt<'tcx>, &mut QueryKeyStringCache)
+        ] = &[$(query_impl::$name::alloc_self_profile_query_strings),*];
+
+        const ENCODE_QUERY_RESULTS: &[
```

Comment:
Could you add a comment saying that those arrays are not indexed per `DepKind`, so should only be used for iteration?